lib/graphql/type/schema.ex

```diff
@@ -81,6 +82,19 @@ defmodule GraphQL.Schema do
     end
   end
 
+  defp reduce_types(typemap, %Input{} = type) do
+    if Map.has_key?(typemap, type.name) do
+      typemap
+    else
+      typemap = Map.put(typemap, type.name, type)
+      thunk_fields = CompositeType.get_fields(type)
+      typemap = Enum.reduce(thunk_fields, typemap, fn({_,fieldtype},typemap) ->
```

Comment:
Good question. This is a bit of a special case since `typemap` is already defined and it's not used outside of the conditional. We could alternatively remove the `typemap = ` in the beginning of line 91.